connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -41,14 +58,33 @@ public void setUpWithConnectButton(ConnectButton connectButton) {
         connectButton.addButtonStateChangeListener(this);
     }
 
-    private ConnectLocation(GeofenceProvider geofenceProvider, ConnectionApiClient connectionApiClient) {
-        this.connectionApiClient = connectionApiClient;
+    private ConnectLocation(String connectionId, GeofenceProvider geofenceProvider,
+            CredentialsProvider credentialsProvider, ConnectionApiClient connectionApiClient,
+            WorkManager workManager) {
+        this.connectionId = connectionId;
         this.geofenceProvider = geofenceProvider;
+        this.credentialsProvider = credentialsProvider;
+        this.connectionApiClient = connectionApiClient;
+        this.workManager = workManager;
```

Comment:
I think we should set up polling when initializing this class as well: for use cases outside of the ConnectButton, we should make sure we set up geofences as soon as possible when the app starts.

Reply:
Added..